soc/espressif/Kconfig.ulp

```diff
@@ -0,0 +1,128 @@
+# Copyright (c) 2025 Espressif Systems (Shanghai) Co., Ltd.
+# SPDX-License-Identifier: Apache-2.0
+
+DT_LPCORE_PARTITION := $(dt_nodelabel_path,lpcore_partition)
+
+menu "Ultra Low Power (ULP) Co-processor"
+	config ULP_COPROC_ENABLED
+		bool "Ultra Low Power (ULP) Co-processor"
+		default n
+		help
+		  Enable this feature if you plan to use the ULP Co-processor.
+		  Once this option is enabled, further ULP co-processor configuration will appear in the menu.
+
+	choice ULP_COPROC_TYPE
+		prompt "ULP Co-processor type"
+		depends on ULP_COPROC_ENABLED
+		default ULP_COPROC_TYPE_LP_CORE if SOC_SERIES_ESP32C6
+		help
+		  Choose the ULP Coprocessor type: ULP FSM (Finite State Machine) or ULP RISC-V.
+
+	config ULP_COPROC_TYPE_FSM
+		bool "ULP FSM (Finite State Machine)"
+		depends on SOC_SERIES_ESP32 || SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+	config ULP_COPROC_TYPE_RISCV
+		bool "ULP RISC-V"
+		depends on SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+	config ULP_COPROC_TYPE_LP_CORE
+		bool "LP core RISC-V"
+		depends on SOC_SERIES_ESP32C6
+	endchoice
+
+	config ULP_COPROC_RESERVE_MEM
+		int
+		prompt "RTC slow memory reserved for coprocessor"
+		depends on ULP_COPROC_ENABLED
+		default $(dt_node_reg_size_int,$(DT_LPCORE_PARTITION))
+		range 32 8176 if SOC_SERIES_ESP32 || SOC_SERIES_ESP32S2 || SOC_SERIES_ESP32S3
+		range 32 16352 if SOC_SERIES_ESP32C6
+		help
+		  Bytes of memory to reserve for ULP Co-processor firmware & data.
+		  Data is reserved at the beginning of RTC slow memory.
+
+menu "ULP RISC-V Settings"
+depends on ULP_COPROC_TYPE_RISCV
+
+	config ULP_RISCV_INTERRUPT_ENABLE
+		bool
+		prompt "ULP RISC-V interrupts"
+		default n
+		help
+		  Turn on this setting to enabled interrupts on the ULP RISC-V core.
+
+	config ULP_RISCV_UART_BAUDRATE
+		int
+		prompt "Baudrate used by the bitbanged ULP RISC-V UART driver"
+		default 9600
+		help
+		  The accuracy of the bitbanged UART driver is limited, it is not
+		  recommend to increase the value above 19200.
+
+	config ULP_RISCV_I2C_RW_TIMEOUT
+		int
+		prompt "Set timeout for ULP RISC-V I2C transaction timeout in ticks."
+		default 500
+		range -1 4294967295
+		help
+		  Set the ULP RISC-V I2C read/write timeout. Set this value to -1
+		  if the ULP RISC-V I2C read and write APIs should wait forever.
+		  Please note that the tick rate of the ULP co-processor would be
+		  different than the OS tick rate of the main core and therefore
+		  can have different timeout value depending on which core the API
+		  is invoked on.
+endmenu
+
+	config ULP_SHARED_MEM
+		depends on ULP_COPROC_TYPE_LP_CORE
+		hex
+		default 0x10
+		help
+		  Size of the shared memory defined in ulp_lp_core_memory_shared.c.
+		  Size should be kept in-sync with the size of the struct defined there.
+
+menu "ULP Debugging Options"
+	config ULP_PANIC_OUTPUT_ENABLE
+		depends on ULP_COPROC_TYPE_LP_CORE
+		bool
+		prompt "Panic handler outputs to LP UART"
```

Comment:
there is no need for prompt on these, combine it with the type line e.g. `bool "Panic handler outputs to LP UART"`